### PR TITLE
[CARBONDATA-4035]Fix MV query issue with aggregation on decimal column

### PR DESCRIPTION
**Pull request by akashrn5**
### Why is this PR needed?
 When the aggregation like sum is performed on decimal column similar to MV schema query, the user query is not getting results from MV. Since in MV schema during MV created we prepare the plan on the query string of modular plan(modularPlan.asCompactSQL) which adds multiple cast expression, because of which the semantic equals fails 
 
 ### What changes were proposed in this PR?
During MV creation, to set the query to MV schema, just use the direct query string, instead of compactSQL.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4434/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2681/

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala [hunk 1]
```diff
@@ -1471,6 +1471,16 @@ class MVCreateTestCase extends QueryTest with BeforeAndAfterAll {
     sql("drop table if exists t2")
   }
 
+  test("test sum aggregations on decimal columns") {
+    sql("drop table if exists sum_agg_decimal")
+    sql("create table sum_agg_decimal(salary1 decimal(7,2),salary2 decimal(7,2),salary3 decimal(7,2),salary4 decimal(7,2),empname string) stored as carbondata")
+    sql("drop materialized view if exists decimal_mv")
+    sql("create materialized view decimal_mv as select empname, sum(salary1 - salary2) from sum_agg_decimal group by empname")
+    sql("explain select empname, sum( salary1 - salary2) from sum_agg_decimal group by empname").show(false)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala:1479]: can revert this change

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala [hunk 2]
```diff
@@ -1471,6 +1471,16 @@ class MVCreateTestCase extends QueryTest with BeforeAndAfterAll {
     sql("drop table if exists t2")
   }
 
+  test("test sum aggregations on decimal columns") {
+    sql("drop table if exists sum_agg_decimal")
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala:1475]: can add drop to afterAll also

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala (hunk 1 above)
Comment (akashrn5) [reply integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala]: done

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala (hunk 2 above)
Comment (akashrn5) [reply integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala:1478]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2877/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4633/

**Comment by Indhumathi27** [comment]
LGTM
